torch/csrc/jit/frontend/lexer.h

```diff
@@ -190,7 +190,7 @@ struct TORCH_API SharedParserData {
   // find the longest match of str.substring(pos) against a token, return true
   // if successful filling in kind, start,and len
   bool match(
-      c10::string_view str,
+      StringCordView str,
```

Comment:
In theory we shouldn't need to optimize memory foorprint inside `lexer` because its source text should go out of scope once compilation is completed.

Same for other use cases like onnx converter, python tree view etc.

Could we verify if this is the case? If it is, maybe we can hide `StringCordView` as an implementation detail behind `source`?

Reply:
> maybe we can hide StringCordView as an implementation detail behind source?

This can be done if Source implements methods such as substr, find directly; then we forward the call to StringCordView. At that point; Source becomes StringCordView. Then, match need to take Source as input instead of StringCordView (still cannot take c10::string_view as input, and we are really in the same spot).